Feature/moderators

softozhik opened the pull request:
много файлов изменено, добавлен рест-контролер, новая роль, новый пользователь, новая таблица в бд, дополнен инициализатор, изменена таблица commentreport и т.д.

softozhik committed:
добавление пункта профиль на страницы модератора

softozhik committed:
модальные окна открываются

softozhik committed:
не сохранялись изменения из-за типа кнопки сохранения в moderator-profile.html, тип изменен на submit - теперь все ок

softozhik committed:
добавлено поле для связи с пользователем, оставившим жалобу в классе ReportComment.java и его dto, а также поле таблицы в moderator-comments(пока не показывает логин)

softozhik committed:
Merge branch 'dev' into feature/moderators

softozhik committed:
вместо объекта пользователя в жалобе сохраняется email

softozhik committed:
в таблицу о жалобах добавлено отображение email пожаловавшегося,
по супермодератору: добавлена ссылка и страница moderator-statistic.html, в инициализатор добавлено создание аккауна supermoderator@mail.ru с паролем 1 и ролью супермодератор, а также, добавки в SecurityConfig и LoginSuccessHandler, создана таблица ModeratorsStatistic с информацией о статистике

softozhik committed:
насоздавал сервисных классов для статистики

softozhik committed:
увеличение счетчиков в статистике

softozhik committed:
табличка со статистикой выводится, но нужно доработать

softozhik committed:
табличка доработана, отображается, как хотел

softozhik committed:
Merge branch 'dev' into feature/moderators

# Conflicts:
#	src/main/java/com/jm/online_store/config/DataInitializer.java
#	src/main/java/com/jm/online_store/config/security/SecurityConfig.java

softozhik committed:
добавил документации

softozhik committed:
мелкие правки

softozhik commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Модераторы](https://trello.com/c/KkevXTZP/227-%D0%BC%D0%BE%D0%B4%D0%B5%D1%80%D0%B0%D1%82%D0%BE%D1%80%D1%8B)

MaksimovMA commented on src/main/java/com/jm/online_store/config/DataInitializer.java:
```diff
@@ -178,12 +181,15 @@ private void roleInit() {
         Optional<Role> servRole = roleService.findByName("ROLE_SERVICE");
         Optional<Role> modrRole = roleService.findByName("ROLE_MODERATOR");
+        Optional<Role> supermodrRole = roleService.findByName("ROLE_SUPERMODERATOR");
+
```
Достаточно одного пропуска строки

softozhik replied:
спасибо, убрал

MaksimovMA commented on src/main/java/com/jm/online_store/config/DataInitializer.java:
```diff
@@ -155,19 +155,22 @@ private void roleInit() {
         Role managerRole = new Role("ROLE_MANAGER");
         Role serviceRole = new Role("ROLE_SERVICE");
         Role moderatorRole = new Role("ROLE_MODERATOR");
+        Role supermoderatorRole = new Role("ROLE_SUPERMODERATOR");
```
camel case мне кажется был бы тут уместен superModeratorRole 
и ROLE_SUPER_MODERATOR для удобства чтения

softozhik replied:
предполагается, что это одно слово, а не два. и дело вкуса. поэтому... пусть ментор решает, переделать или нет :-)

MaksimovMA commented on src/main/java/com/jm/online_store/controller/rest/moderator/ModeratorRestController.java:
```diff
@@ -55,7 +61,9 @@
         reportComment.setComment(commentService.findById(reportCommentDto.getCommentId()));
+        reportComment.setReportCustomerEmail(userService.getCurrentLoggedInUser().getEmail());
         reportCommentService.addReportComment(reportComment);
+//        System.out.println();
```
Зачем?

softozhik replied:
спасибо, убрал

softozhik committed:
модельное окно для смены картинки профиля заработало

softozhik committed:
поправил путь к каталогу с картинками в UserServiceImpl.java, поправил модалки в профиле manager-а

softozhik committed:
убрал немного мусора

softozhik committed:
return-ы поправил с учетом нового пути к папке с рисунками

softozhik committed:
вернул назад пути к файлам картинок

для работы нужно, чтобы папка uploads/images/ была в корне проекта, если ее нет - то создать или скопировать из resource, т.к. сама папка в gitignore, т.е. не обновляется при синхронизации с репозиторием

softozhik committed:
вернул назад пути к файлам картинок

для работы нужно, чтобы папка uploads/images/ была в корне проекта, если ее нет - то создать или скопировать из resource, т.к. сама папка в gitignore, т.е. не обновляется при синхронизации с репозиторием

softozhik commented:
вернул назад пути к файлам картинок

**!!!ВАЖНО!!!**
**для работы нужно, чтобы папка uploads/images/ была в корне проекта, если ее нет - то создать или скопировать из resource, т.к. сама папка в gitignore, т.е. не обновляется при синхронизации с репозиторием**

tediik commented on src/main/java/com/jm/online_store/controller/rest/moderator/ModeratorRestController.java:
```diff
@@ -79,7 +87,20 @@
+     * Вывод статистики работы с жалобами
+     *
+     * @return List<ModeratorsStatistic>
+     */
```
Лишняя пустая строка между комментариями НЕ нужна, используем только когда текста много, чтобы отделить от описания параметров.
Распиши комметарий к возвращаемому значению.

tediik commented on src/main/java/com/jm/online_store/controller/rest/moderator/ModeratorRestController.java:
```diff
@@ -79,7 +87,20 @@
+     * @return List<ModeratorsStatistic>
+     */
+    @GetMapping("/statistic")
+    @ApiOperation(value = "List moderators activities")
```
почему в названии и мапинге статистика, а в описании к апи появились какие то активности? Не надо путать людей и мешать понятия, придерживайся единообразия в коде

tediik commented on src/main/java/com/jm/online_store/controller/simple/ModeratorController.java:
```diff
@@ -23,4 +27,24 @@ public String moderatorPage(Model model) {
+    @GetMapping("/profile")
+    public String getPersonalInfo(Model model) {
+        User user = userService.getCurrentLoggedInUser();
+        model.addAttribute("user", user);
```
больше ничего не отдаем на фронт через таймлиф и обычные контроллеры, со страницы будет запрос на соответствующий рест-контроллер чтобы получить текущего залогиненного пользователя

tediik commented on src/main/java/com/jm/online_store/model/ReportComment.java:
```diff
@@ -15,11 +15,13 @@
  * Сущность жалоб на {@link Comment}.
  */
 @Entity
+@Table(name = "report_comment")
```
ну оно в общем и так же работало, аннотация не является обязательной

tediik commented on src/main/java/com/jm/online_store/repository/ModeratorsStatisticRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ModeratorsStatisticRepository extends JpaRepository<ModeratorsStatistic, Long> {
+    ModeratorsStatistic getByModeratorEquals(User moderator);
```
лучше - getModeratorsStatisticByModerator

tediik commented on src/main/java/com/jm/online_store/service/impl/ModeratorsStatisticServiceImpl.java:
```diff
@@ -0,0 +1,74 @@
+
+    private final ModeratorsStatisticRepository moderatorsStatisticRepository;
+
+    /**
+     * список статистики отклоненных и принятых жалоб
+     *
+     * @return List<ModeratorsStatistic>
+     */
```
во всем классе убери пустую разделяющую строку в комментариях, пожалуйста

tediik commented on src/main/resources/static/js/moderator-comments.js:
```diff
@@ -83,6 +85,32 @@ function renderCommentsTable(allReports) {
+                        let firstName = (s.moderator.firstName == null) ? `` : s.moderator.firstName;
+                        let lastName = (s.moderator.lastName == null) ? `` : s.moderator.lastName;
+
+                        statistic += `<tr>`;
+                        statistic += `<th>` + firstName + ` ` + lastName + `</th>`;
+                        statistic += `<th>` + s.moderator.email + `</th>`;
+                        statistic += `<th>` + s.approvedCount + `</th>`;
+                        statistic += `<th>` + s.dismissedCount + `</th>`;
+                        statistic += `<th>` + (s.approvedCount + s.dismissedCount) + `</th>`;
+                        statistic += `<th>` + s.lastActivityDate + `</th></tr>`;
+                        }
+                    )
```
это неправильное использование шаблонных строк js, пересмотри лекции

tediik commented on src/main/resources/templates/moderator-profile.html:
```diff
@@ -0,0 +1,305 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org" xmlns:sec="http://www.w3.org/1999/xhtml" lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Profile</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+
```
а у нас разве не было странички модератора??

tediik commented on src/main/resources/templates/moderator-statistic.html:
```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org" xmlns:sec="http://www.w3.org/1999/xhtml">
+
+<head>
+    <title>Moderator Page</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
```
выше ты добавил профиль модератора, тут страница статистики, но при этом тут тоже меню кабинета модератора?

softozhik committed:
почти все замечания исправлены, есть недочеты - не отображается информация по ссылке при первом открытии страницы и с изменением профиля тоже пока только через thymeleaf

softozhik committed:
зараработала табличка комментариев для проверки и количество комментариев, ожидающих проверки

softozhik committed:
внезапно заработал готовый профиль profile-admin-manager-service.html, не понимаю, почему раньше не подключался

softozhik committed:
доделал, чтобы не было необходимости передавать пользователя через обычный контролер

softozhik committed:
id для удаления пользователя беру из js, а не html

softozhik committed:
удалить картинку

softozhik committed:
удалил кнопку удаления профиля в profile-admin-manager-service.html

softozhik committed:
удалил функционал для кнопки удаления профиля

softozhik committed:
поправил - из другого рест-контролера удаление профиля

softozhik committed:
неиного мусора убрал

softozhik committed:
Merge branch 'dev' into feature/moderators

softozhik committed:
вернул обратно метод в рест-контролере

softozhik committed:
Merge branch 'dev' into feature/moderators

# Conflicts:
#	src/main/java/com/jm/online_store/config/security/SecurityConfig.java

softozhik committed:
решение конфликта в SecurityConfig.java

softozhik committed:
убрал лишние пустые строки в ModeratorsStatisticServiceImpl
